redbot/cogs/mutes/mutes.py

```diff
@@ -0,0 +1,1431 @@
+import asyncio
+import contextlib
+import discord
+import logging
+
+from abc import ABC
+from copy import copy
+from typing import cast, Optional, Dict, List, Tuple, Literal, Union
+from datetime import datetime, timedelta, timezone
+
+from .converters import MuteTime
+from .voicemutes import VoiceMutes
+
+from redbot.core.bot import Red
+from redbot.core import commands, checks, i18n, modlog, Config
+from redbot.core.utils.chat_formatting import humanize_timedelta, humanize_list, pagify
+from redbot.core.utils.mod import get_audit_reason
+from redbot.core.utils.menus import start_adding_reactions
+from redbot.core.utils.predicates import MessagePredicate, ReactionPredicate
+
+T_ = i18n.Translator("Mutes", __file__)
+
+_ = lambda s: s
+
+MUTE_UNMUTE_ISSUES = {
+    "already_muted": _("That user is already muted in this channel."),
+    "already_unmuted": _("That user is not muted in this channel."),
+    "hierarchy_problem": _(
+        "I cannot let you do that. You are not higher than the user in the role hierarchy."
+    ),
+    "is_admin": _("That user cannot be muted, as they have the Administrator permission."),
+    "permissions_issue": _(
+        "Failed to mute or unmute user. I need the manage roles "
+        "permission and the user I'm muting must be "
+        "lower than myself in the role hierarchy."
+    ),
+    "left_guild": _("The user has left the server while applying an overwrite."),
+    "unknown_channel": _("The channel I tried to mute or unmute the user in isn't found."),
+    "role_missing": _("The mute role no longer exists."),
+    "voice_mute_permission": _(
+        "Because I don't have the Move Members permission, this will take into effect when the user rejoins."
+    ),
+}
+_ = T_
+
+log = logging.getLogger("red.cogs.mutes")
+
+__version__ = "1.0.0"
+
+
+class CompositeMetaClass(type(commands.Cog), type(ABC)):
+    """
+    This allows the metaclass used for proper type detection to
+    coexist with discord.py's metaclass
+    """
+
+    pass
+
+
+class Mutes(VoiceMutes, commands.Cog, metaclass=CompositeMetaClass):
+    """
+    Mute users temporarily or indefinitely.
+    """
+
+    def __init__(self, bot: Red):
+        self.bot = bot
+        self.config = Config.get_conf(self, 49615220001, force_registration=True)
+        default_guild = {
+            "sent_instructions": False,
+            "mute_role": None,
+            "notification_channel": None,
+            "muted_users": {},
+            "default_time": {},
```

Comment:
This default needs to be updated, here's one of errors I get due to current default value:
```
[2020-10-25 00:04:11] [ERROR] red: Exception in command 'muteset settings'
Traceback (most recent call last):
  File "C:\Users\Jakub\OneDrive\Dokumenty\Red-DiscordBot\.venv\lib\site-packages\discord\ext\commands\core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "C:\Users\Jakub\OneDrive\Dokumenty\Red-DiscordBot\redbot\cogs\mutes\mutes.py", line 530, in show_mutes_settings
    default_time = timedelta(seconds=data["default_time"])
TypeError: unsupported type for timedelta seconds component: dict
```

Cog seems to be working fine when `0` is set as the default:
```suggestion
            "default_time": 0,
```